extension/json/json_functions/json_pretty.cpp

```diff
@@ -0,0 +1,32 @@
+#include "json_executors.hpp"
+
+namespace duckdb {
+
+//! Pretty Print a given JSON Document
+string_t PrettyPrint(yyjson_val *val, yyjson_alc *alc, Vector &result) {
+	D_ASSERT(alc);
+	idx_t len;
+	auto data =
+	    yyjson_val_write_opts(val, JSONCommon::WRITE_PRETTY_FLAG, alc, reinterpret_cast<size_t *>(&len), nullptr);
+	return StringVector::AddString(result, data, len);
```

Comment:
You can simply return `string_t(data, len)` here, as the string will stay cached for long enough in the `alc` until the next vector is processed. Using `StringVector::AddString` here will unnecessarily create an extra copy of the string.

Reply:
Ah. I had that before (https://github.com/duckdb/duckdb/pull/12398#discussion_r1627980229). So I should roll back that change?

Reply:
Yes, it can be rolled back. We don't have to add the string to the vector since it will be cached in the allocator. We're doing the exact same thing in `json_extract.cpp` and `json_transform.cpp`. Sorry for the confusion.